packages/wallet/src/components/dialogs/SuccessDialog.tsx

```diff
@@ -59,25 +61,28 @@ export const SuccessDialog = ({
                   </div>
                 </div>
                 <div className="mt-5 grid grid-cols-1 gap-3">
-                  <Button
-                    intent="success"
-                    aria-label="close dialog"
-                    fullWidth
-                    onClick={() => onClose()}
-                  >
-                    Close
-                  </Button>
                   {redirect && (
                     <Button
                       intent="secondary"
                       aria-label={redirectText ?? 'redirect'}
                       fullWidth
                       href={redirect ?? '/'}
-                      onClick={() => onClose()}
+                      onClick={() => {
+                        onSuccess ? onSuccess() : undefined
+                        onClose()
+                      }}
                     >
                       {redirectText}
                     </Button>
                   )}
+                  <Button
+                    intent="success"
+                    aria-label="close dialog"
+                    fullWidth
+                    onClick={() => onClose()}
+                  >
+                    Close
+                  </Button>
```

Comment:
🤔 I don't think we need the close button. Clicking it, it will just close the dialog and not redirect the user to the next tab.